lib/daru/maths/statistics/vector.rb

```diff
@@ -855,6 +835,14 @@ def raw_sample_without_replacement sample
 
           out.collect { |i| valid[i] }
         end
+
+        def sort_without_nil(&block)
+          dv = reject_values(nil)
+
+          return dv.sort.to_a unless block_given?
+          return dv.sort_by(&block) if block.parameters.count == 1 # Object block like { |x| x.size }
+          dv.sort(&block).to_a # Comparative block like { |a,b| a.size <=> b.size }
```

Comment:
Acknowledged. Changed `Daru::Vector#max` and `Daru::Vector#max_by` and other similar methods, to resemble the usual enum `#feature` and `#feature_by` behaviour.